include/service/entities/plugin_service.js

```diff
@@ -538,6 +538,65 @@ module.exports = function PluginServiceModule(pb) {
      * existing settings for the plugin are deleted before the new settings are
      * persisted.
      *
+     * @method syncSettings
+     * @param details The details object to extract the settings from
+     * @param cb A callback that provides two parameters: cb(error, TRUE/FALSE).
+     * TRUE if the settings were successfully cleared and reloaded. FALSE if not.
+     */
+    PluginService.prototype.syncSettings = function(plugin, details, cb) {
+        var self = this;
+        this.getSettingsKV(plugin.dirName, function(err, settings) {
+            if (pb.util.isError(err) || !settings) {
+                pb.log.error("PluginService.syncSettings failed to load settings from plugin [" + plugin.dirName + "]");
+                return cb(null, true);
+            }
+            var discrepancy = false;
+            var hashedSettings = {};
+            var formattedSettings = [];
+
+            // Detect new settings
+            for (var i = 0; i < details.settings.length; i++) {
+                var settingName = details.settings[i].name;
+                hashedSettings[settingName] = true;
+                if (settings[settingName] === undefined) {
+                    discrepancy = true;
+                    formattedSettings.push({name: details.settings[i].name, value: details.settings[i].value});
+                }
+                else {
+                    formattedSettings.push({name: settingName, value: settings[settingName]});
+                }
+            }
+
+            // Detect deprecated settings if we haven't already detected a discrepancy
+            if (!discrepancy) {
+                for (var setting in settings) {
+                    if (!hashedSettings[setting]) {
+                        discrepancy = true;
+                        break;
+                    }
+                }
+            }
```

Comment:
We brainstormed methods of detecting discrepancies for a good long while and these ~2 loops seemed the most optimized but I'm open to alternatives if you can think of one.